app/src/main/java/fr/free/nrw/commons/contributions/RxJava2Tasks.java

```diff
@@ -0,0 +1,30 @@
+package fr.free.nrw.commons.contributions;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Locale;
+
+import fr.free.nrw.commons.PageTitle;
+import io.reactivex.Single;
+
+class RxJava2Tasks {
+
+    private static final String UPLOAD_COUNT_URL_TEMPLATE =
+            "https://tools.wmflabs.org/urbanecmbot/uploadsbyuser/uploadsbyuser.py?user=%s";
+
+    static Single<Integer> getUploadCount(String userName) {
```

Comment:
The time investment for automated tests wouldn't be to prove that the legacy `ApacheHttpClientMediaWikiApi` was doing things right as much as to give ourselves a safety net for the upgrade to come.  In that sense it's perfectly valid to do the work later!

🤣 I've been told that I "code like a machine" but it's the first time someone's called my code auto-generated 😸